lib/install/management/delete.go

```diff
@@ -342,3 +354,53 @@ func (d *Dispatcher) networkDevices(vmm *vm.VirtualMachine) ([]types.BaseVirtual
 	}
 	return devices, nil
 }
+
+func (d *Dispatcher) deleteFolder() {
+	var err error
+
+	// no inventory folders if we are targeting ESX
+	if !d.isVC {
+		return
+	}
+
+	d.op.Info("Removing VCH Inventory Folder")
+	vchFolderPath := path.Dir(d.appliance.InventoryPath)
+	defer func() {
+		if err != nil {
+			d.op.Warnf(manualInventoryCleanWarning, vchFolderPath)
+		}
+	}()
+
+	folderRef, err := d.session.Finder.Folder(d.op, vchFolderPath)
+	if err != nil {
+		d.op.Debugf("failed to find folder: %s for the VCH target", vchFolderPath)
+	}
+
+	// protections against old vch's since they are directly in the VMFolder
+	dcFolder, err := d.session.Datacenter.Folders(d.op)
+	if err != nil {
+		d.op.Debugf("failed to find the datacenter folders: %s ", err)
+	}
+	VMFolder := dcFolder.VmFolder
+	if folderRef.Reference() == VMFolder.Reference() {
+		// cannot delete the vm folder
+		return
+	}
+
+	// we need to see if the folder is empty.
+	folderContents, err := folderRef.Children(d.op)
+	if err != nil || len(folderContents) != 0 {
+		d.op.Debugf("Could not remove VCH inventory folder, %s has existing contents in it. Manual cleanup required.", vchFolderPath)
+	}
+
+	// NOTE: Destroy on Inventory Folders is RECURSIVE, start from the leaf most target and check folder children to avoid undesired deletions.
+	folderRemoveFunction := func(ctx context.Context) (tasks.Task, error) {
+		return folderRef.Destroy(d.op)
+	}
+
+	_, err = tasks.WaitForResult(d.op, folderRemoveFunction)
```

Comment:
I prefer this approach:
```
info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
		return vchFolder.CreateVM(ctx, *spec, d.vchPool, d.session.Host)
	})
``` 
is that not applicable here?  I'm not seeing the reason for the func var...